### DISCUSSION TITLE
Does anyone have sucessfully setup Facebook Channel ? issue with Access Verification and Control App ?

### DISCUSSION THREAD

**Discussion by slamer59** [chatwoot/chatwoot#8776] [Q&A]
Hello,
it is now a week that I tried to put chatwoot to production on a self hosted instance. 

If everything is working technically for stagin, I fail misarbily to :
- Convice the controler ont the necessity of all API
- Get verified access and prove that Jokosun is a provider

# Control App part
I tried 4 times to for the Control App : 
Here is one out of many answer from the controler : 
> While reviewing this submission, we were unable to verify the app because the use case is unclear.
> Specifically, we noticed that your use case details is not valid for how the permission is supposed to be used.
> The pages_show_list permission allows your app to access the list of Pages a person manages. The allowed usage for this permission is to show a person the list of Pages they manage and verify that a person manages a page.
> The allowable use cases for this permission are the following:
> Show a person the list of Pages they manage
> Publish a post, photo and or video (including Reels video) to their FB page
> 
> Pages_show_list is considered disallowed if it access the detailed information of:
> Page Posts
> Page Photos
> Page Albums
> Page Videos
> Page Live Videos
> Page Comments
> Access Tokens to the Page/Page Posts
> 
> For the next submission, please provide us with a verifiable use case for the review team to test and verify the functionality of the requested permission.
> 
> You may refer to this link for more information https://developers.facebook.com/docs/permissions/reference
> Please re-submit accordingly and we will re-review your application.
> 
> Thank you.

I follow exactly this 2 years old video : 
https://www.youtube.com/watch?v=BnAGfjvIUVU

# Documentation to be enhanced ?

The documentation doesnot mention the access approval and could give hints on how to pass production in facebook (Even if it is not a science). 
https://www.chatwoot.com/docs/self-hosted/configuration/features/integrations/facebook-channel-setup/

# Facebook Access verification procedure
https://developers.facebook.com/docs/development/release/access-verification/

> This is the rejection message from FB (translated) : 
> Access check was rejected
>  
>  
> We have reviewed your responses for the access verification of your Jokosun company. Unfortunately, we cannot approve your business because we have determined that it does not operate as a technology provider. API calls to certain permissions and features (advanced access) for the apps listed below will experience an interruption.
> 
> If you disagree with this decision, an administrator from your company can carry out the access check again.

This is the request I have filled (translated) : 

> **Which of the following best describes your business? Select all possible answers.**
> If your business has multiple companies or brands, select the option that best describes the division of your business that provides services as a technology provider.
> - Agency
> - Freelance
> - Consulting
> - SaaS platform
> - [X] Other
> We are a solar company and we want to interact with users
> **How will your business use Platform Data (i.e. any information or data you obtain from us) to activate a product or service on behalf of your customers?**
> **How will your business use Platform Data (i.e. any information or data you obtain from us) to activate a product or service on behalf of your customers?**
> The use of data from the Chatwoot platform by our company will be done in strict compliance with confidentiality and information security. The data we obtain from the platform will be used exclusively for the purpose of activating products or services on behalf of our customers. This can include personalizing user interactions, improving customer experience, and providing more targeted and efficient services.
> 
> We are committed to treating data with the utmost confidentiality and implementing robust security measures to protect the integrity and confidentiality of information. Under no circumstances will we share or sell the data to third parties without the explicit consent of our customers.
> 
> Our use of platform data will be aligned with ethical principles and applicable data protection regulations. We seek to maximize benefits for our customers while ensuring responsible and transparent use of information obtained through Chatwoot.
> 
> Does your business manage multiple Business Accounts?
> No
> 
> Provide a link to your website.
> https://www.jokosun.com

# Questions ?
I really like the idea of having a central hosted solution (and cheap) but I didn't see how I would solve this issue if it is not going into production. Does anyone has successfully registered facebook app recently (less than a year) ?
Can we share experience ? 
What cheap alternatives do we have (less than 10€/month would be ideal) ? 

I would be glad to enhance the documentation in case of success. I can also share videos for Control App if required.

**Comment by sojan-official** [comment]
https://github.com/orgs/chatwoot/discussions/3397
